### DISCUSSION TITLE
Media playback

### DISCUSSION THREAD

**Discussion by rekkyrosso** [tidal-music/discussions#8] [Ideas & Feature requests]
I would like to be able to play TIDAL content in a separate web app. Like this:

https://ampcast.app/

Ability to play both audio and video would be nice.

It would also be good if the playback was not enabled through an iframe so that we can use an AnalyserNode to create visualisers.

https://developer.mozilla.org/en-US/docs/Web/API/AnalyserNode

**Comment by stefanmoro** [comment]
Hi @rekkyrosso. Welcome to TIDAL discussions and thanks for your input and feedback.
 
Media playback is one of the features we expect to be most requested by devs building on TIDAL Developer Platform, and it is something we are currently investigating and working on.
 
There are several aspects to this work, and we hope to open up this functionality in stages. In addition to working on the technical aspects of how to enable this functionality, we are also working together with our label partners to find a solution that is beneficial for all involved parties.
 
Stay tuned and we'll keep you posted once we have any updates regarding media playback!
 
Thanks

**Reply by rekkyrosso** [reply]
A year later and the SDK is really starting to take shape.

I can search, browse the catalogue and play a preview of tracks. So great work and excellent progress!

Can you give us an ETA for full playback? It would be great to finally play full tracks.

**Reply by stefanmoro** [reply]
Hi @rekkyrosso, thanks for reaching out.

We aim to open up for full-length playback through the SDK at the same time as we open up for "production mode requests". We want the API and SDK to stabilize a bit and leave the "beta" phase before we do that. I hope we'll be able to reach that state by the end of this year.

**Reply by pierrep** [reply]
@stefanmoro where will this be announced (playback support) when it happens? I'm creating my own open source audio playback tool for live sound design, and would be super keen to add Tidal integration at some stage.

**Reply by stefanmoro** [reply]
Hey @pierrep ,

We will announce it in this discussion forum.

**Reply by xplosiveloons** [reply]
Hey @stefanmoro any updates on the status of full media playback with the SDK?

**Reply by aligator** [reply]
I just tried my code from some time ago and it seems to work now. At least could I play a full song. Didn't test much more yet.

Edit: nevermind, it did only play a full song one time. Now it plays the previews again.
Edit2: strange - I now started the tidal (linux) desktop app on the same PC (just starting, without playing) and then playing a full song magically works again from my dev-webapp... How is that possible?

**Reply by r3d12** [reply]
so far im not able to do full playback in a web app with the web sdk. unless im missing something

**Comment by thephenakist** [comment]
To help elaborate on a use case scenario, I have a socket-based application which is essentially a glorified chat server. Server-side we broadcast messages to all of the users in a particular room with the ID of the track to play, and how far into the track to start playing (i.e. consider a user who joins the room in the middle of a song). The clients then call in to initiate playback (in this case it would be within TIDAL's ecosystem) of the specified audio file.

The hope is to have API endpoints which allow a user to _list_ and _interact_ with their TIDAL audio devices. For example:
- Get list of TIDAL audio devices
- Start / Resume playback of a track at a given offset, in milliseconds (i.e. `playTrack('75623239', 31502)` to start the track at 31.5s)
- Set playback volume for current audio device

The above three items are most valuable for my purposes, but it stands to reason that anything the user can currently do within the current web player, they would like to be able to accomplish (toggle repeat mode, next/prev track, get details about the currently playing song, play audio on multiple devices simultaneously vis-a-vis Airplay 2, add/remove items from the queue, view recently played tracks, etc).

**Comment by HappyKonst** [comment]
@stefanmoro is it now possible to play full tracks with tidal web sdk if someone is authenticated with a paid tidal account?

**Reply by lo92fr** [reply]
Can you give answer on the last question.
What are the release timeline for a full playback on the web sdk.
We are now waiting for really long time on it.

**Comment by ddingel** [comment]
I wanted to see how hard it is to built a bridge for Hitster with Tidal as a provider, comparable to: https://www.hitster4music.com. 

For this it would be really nice to be able to play good quality full length audio tracks.

**Reply by rjelierse** [reply]
Very early stage, mostly as an exercise to be able to play the Hitster versions I have at home, but it works https://ripster.jelierse.net/
